### DISCUSSION TITLE
Qwen 0.5B Results

### DISCUSSION THREAD

**Discussion by thomasj02** [VsonicV/es-fine-tuning-paper#35] [General]
In the text for Figure 1 in the paper, I see that it says 
> For instance, Pan et al. (2025) note that “for Qwen2.5-0.5B base, we know
it fails to learn reasoning.” Surprisingly, ES overcomes this limitation. As shown in Table 1, while
PPO and GRPO indeed obtain only 0.3% accuracy on that model, ES boosts accuracy to 14.4%, thus
eliciting reasoning even from the smallest-scale base model

I'm currently running the code with Qwen2.5-0.5B base. The run is ongoing, but it's currently at about 17% accuracy on the training data. Still, I think it seems clear that it's not learning reasoning. Here's a representative output:

> {"iteration": 211, "seed_idx": 29, "sample_idx": 193, "input": "You are a helpful assistant. You first think about the reasoning process in your mind and then provide the user with the
 answer.Using the numbers [78 54 52], create an equation that equals 76. You can use basic arithmetic operations (+, -, *, /) and each number can only be used once. Show your work in <
think> </think> tags. And return the final answer in <answer> </answer> tags, for example <answer> (1 + 2) / 3 </answer>.Let me solve this step by step.\n<think>", "target": "76", "res
ponse": "You are a helpful assistant. You first think about the reasoning process in your mind and then provide the user with the answer.Using the numbers [78 54 52], create an equatio
n that equals 76. You can use basic arithmetic operations (+, -, *, /) and each number can only be used once. Show your work in <think> </think> tags. And return the final answer in <a
nswer> </answer> tags, for example <answer> (1 + 2) / 3 </answer>.Let me solve this step by step.\n<think>78 - 54 + 52</think>\n<answer>(78 - 54) + 52</answer>", "reward": 1.1, "correc
t": true}

You'll notice that the "reasoning" is just `<think>78 - 54 + 52</think>`
Did you get true reasoning in your experiments with Qwen2.5-0.5B-Instruct and I'm getting these results just because I'm using Base like Pan (2025)?

**Comment by VsonicV** [comment]
@thomasj02 Your observation is right. We did notice that Qwen2.5-0.5B-Instruct was very concise in answering even before the fine-tuning. After ES fine-tuning, it was still very concise in answering, but indeed with significantly higher chances to produce correct answers. What we intended to say here is that the model was able to learn to solve this reasoning problem from a ~0% initial accuracy, perhaps performing reasoning implicitly (internally). We agree that "learn reasoning" can also be interpreted as performing reasoning behaviors explicitly in outputs. We will clarify this point in the next version of our paper. Sorry about the confusion. Thanks for pointing this out!
